titus-ext/cassandra/src/main/java/com/netflix/titus/ext/cassandra/store/CassandraLoadBalancerStore.java

```diff
@@ -152,17 +206,11 @@ public void init() {
                         if (failOnError) {
                             throw LoadBalancerStoreException.badData(jobLoadBalancer, violations);
                         }
-                        logger.warn("Ignoring bad record of {} due to validation constraint violations: violations={}", jobLoadBalancer, violations);
+                        logger.warn("Ignoring bad association record of {} due to validation constraint violations: violations={}", jobLoadBalancer, violations);
                     }
-                });
-    }
-
-    @Override
-    public Observable<JobLoadBalancerState> getLoadBalancersForJob(String jobId) {
-        logger.debug("Getting all load balancer states for job {}", jobId);
-        return Observable.from(loadBalancerStateMap.entrySet())
-                .filter(entry -> entry.getKey().getJobId().equals(jobId))
-                .map(JobLoadBalancerState::from);
+                })
+                .ignoreElement()
```

Comment:
Are `ignore()` + `cast()` doing something different than `then()`?

Reply:
Not really, it was on oversight. Good catch.